indigo_app/static/javascript/indigo/views/library.js

```diff
@@ -356,26 +356,20 @@
 
         work = work.toJSON();
 
-        work.docs = _.map(docs[work.id] || [], function(d) { return d.toJSON(); });
-        // latest expression first
-        work.docs = _.sortBy(work.docs, 'expression_date');
-        work.docs.reverse();
+        var work_docs = docs[work.id] || [];
 
-        // current user's name -> you
-        work.docs.forEach(function(d, i) {
-          if (d.updated_by_user && d.updated_by_user.id == currentUserId) d.updated_by_user.display_name = 'you';
+        // distinct languages
+        work.languages = _.unique(_.map(work_docs, function(doc) {
+          return doc.get('language');
+        }));
 
-          // only show doc titles that are different to the work
-          if (i > 0 && d.title == work.title) d.title = '';
-        });
-
-        if (work.docs.length > 0) {
-          var dates = _.compact(_.pluck(work.docs, 'updated_at'));
-          dates.sort();
+        // count expression dates
+        work.expression_dates = (_.unique(_.map(work_docs, function(doc) {
```

Comment:
Nice. You don't need the outermost parentheses.

Entirely a matter of style, I would call this `n_expressions` or similar to call out that it's a count, not a list. But up to you.